components/feature/webnotifications/src/main/java/mozilla/components/feature/webnotifications/NativeNotificationBridge.kt

```diff
@@ -37,22 +43,22 @@ internal class NativeNotificationBridge(
         }
 
         with(notification) {
-            loadIcon(iconUrl?.toUri(), origin, Size.DEFAULT)?.let { icon ->
-                builder.setLargeIcon(icon)
+            activityClass?.let {
+                val intent = Intent(context, activityClass)
+                intent.putExtra(EXTRA_ON_CLICK, tag)
+                val pendingIntent = PendingIntent.getActivity(context, requestId, intent, 0)
+                builder.setContentIntent(pendingIntent)
             }
 
-            builder.setContentTitle(title).setContentText(body)
-
-            @Suppress("Deprecation")
-            builder.setVibrate(vibrate)
+            builder.setSmallIcon(launcherIcon).setContentTitle(title).setContentText(text)
+                    .setAutoCancel(true)
 
-            timestamp?.let {
-                builder.setShowWhen(true).setWhen(it)
+            loadIcon(imageUrl?.toUri(), Size.DEFAULT)?.let { iconBitmap ->
+                builder.setLargeIcon(iconBitmap)
             }
 
-            if (silent) {
-                @Suppress("Deprecation")
-                builder.setDefaults(0)
+            timestamp.let {
```

Comment:
If `timestamp` is always set there's no need for the `let` block

Reply:
Oh yeah, I had timestamp as possible `null` but not anymore.  I'll fix.  Thanks,